_posts/2023-07-13-correlating-security-events.md

```diff
@@ -0,0 +1,123 @@
+
+
+## Why do I need a correlation framework?
+
+As data spans across multiple security event logs (s3 access, VPC flow, sys log, DNS ), a finding on just one log source is not enough to increase the confidence of the finding. A strong correlation across logs helps customers to dive into the relationship of data across different sources.
+
+To understand a correlation between findings from different log sources, you would need to manually browse through a list of generated findings for individual log categories and identify any pattern between the findings manually. 
+
+The following sample customer infrastructure diagram further illustrates the need for a correlation framework. This customer deploys a Django REST application hosted on a EC2 Windows instance. The REST APIs for the application use Active Directory as the identity provider and Amazon S3 to store and query files. The EC2 Windows instance stores incoming network traffic in VPC Flow Logs.
+
+
+<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/customer application.png" alt="Sample customer infrastructure"/>{: .img-fluid }
+
+For Security Analytics to monitor and detects threats for this infrastructure, we need to the define a detector for each component in the infrastructure. The following diagram shows the infrastructure with a detector for each component, such as a network detector for VPC Flow logs and a Windows detector for the EC2 Windows instance.
+
+
+<img src="/assets/media/blog-images/2023-07-13-correlating-security-events/detector-infrastructure.png" alt="Detector infrastructure inside of customer's application"/>{: .img-fluid }
+
+With the detectors set, we can stimulate a security attack on this example infrastructure. The attacker uses `sbcd90` user to call a REST API named `POST /customer_records.txt` which tries to replicate a sensitive file named` customer_records.txt` from S3.
+
+
+When the attack occurs, each detector generates a finding for its corresponding infrastructure component. For example, the AD/LDAP detector generates a finding that indicates `Invalid Username/Password, ResultType: 50126`. This finding belongs to a that individual detectors respective log type. 
+
+But how would you know that the AD/LDAP detector finding `Invalid Username/Password, ResultType: 50126` is related to a chain of security events occurring in the same time window on the infrastructure, such a `403 Forbidden error` finding from an Application logs detector? Is there a way to correlate finding that belong to other log type in a particular time range automatically? 
```

Comment:
If we answer these questions in the following paragraphs, we should say so here.

Reply:
Something like, "With correlation engine, you can." Or "Correlation engine makes all of the above possible." Or, "You can achieve all of these tasks through correlation engine."

Thoughts?

Reply:
I like either the first or second one. If you use the second one, "this" instead of "the above".